lib/rules/no-console.js

```diff
@@ -40,26 +102,52 @@ module.exports = {
         return {
 
             MemberExpression: function(node) {
+                var scope = context.getScope();
+                var passedProperty = getPropertyName(node);
+                var passedMemberExpressionProperty = getPropertyName(node.object);
+                var variable;
+                var propertyIsAllowed;
 
-                if (node.object.name === "console") {
-                    var blockConsole = true;
+                if (context.options.length > 0 && context.options[0].allow) {
+                    propertyIsAllowed = (context.options[0].allow.indexOf(passedProperty) > -1);
+                }
 
-                    if (context.options.length > 0) {
-                        var allowedProperties = context.options[0].allow;
-                        var passedProperty = node.property.name;
-                        var propertyIsAllowed = (allowedProperties.indexOf(passedProperty) > -1);
+                // checks console.*() [* means any console method]
+                if (node.object.name === "console") {
+                    variable = astUtils.getVariableByName(scope, "console");
 
-                        if (propertyIsAllowed) {
-                            blockConsole = false;
-                        }
+                    if (!isShadowed(variable) && !propertyIsAllowed) {
+                        report(context, node);
                     }
 
-                    if (blockConsole) {
-                        context.report(node, "Unexpected console statement.");
+                // checks window.console.*(), self.console.*(), && this.console.*()
+                } else if (passedMemberExpressionProperty === "console") {
+                    if (context.options.length && context.options[0].disallowGlobals) {
+                        var passedObject = node.object.object;
+                        var disallowedGlobals = context.options[0].disallowGlobals;
+
+                        switch (passedObject.name) {
+                            case "window":
```

Comment:
@nzakas @ilyavolodin I am assuming that a user should only be able to pass in global references. So (I may be completely wrong) I thought the only built in references to the global object is `window`, `self`, and `this`. Now that I look at it again that code should absolutely be simplified because both cases are definitely doing the same thing. 

But regardless, I have to check that the items passed into `disallowGlobals`, is actually a valid global right? Or rather, should a user be able to specify, in `disallowGlobals`, some variable that they use that references the global object. For example:
```js
// eslint config
no-console: ["error", { "allow": ["error"], "disallowGlobals": [ "window", "myWindow" ] }]

// their project
var myWindow = this;

myWindow.console.log('hello world'); // should throw an error
```
Sorry for the long write-up. Thanks for your patience.